transport/config/config.go

```diff
@@ -0,0 +1,143 @@
+	// Default false
+	DisableKeepAlives bool `yaml:"DisableKeepAlives"`
+}
+
+// ClientTransportTriggers properties
+type ClientTransportTriggers struct {
+	Method     string `yaml:"Method" validate:"max=64"`
+	Path       string `yaml:"Path" validate:"max=64"`
+	QueryParam string `yaml:"QueryParam" validate:"max=64"`
+}
+
+// TriggersCompiledRules properties
+type TriggersCompiledRules struct {
+	MethodRegexp     *regexp.Regexp
+	PathRegexp       *regexp.Regexp
+	QueryParamRegexp *regexp.Regexp
+	IsCompiled       bool
+}
+
+// Transport properties
+type Transport struct {
+	Name                  string                  `yaml:"Name"`
+	Triggers              ClientTransportTriggers `yaml:"Triggers"`
+	TriggersCompiledRules TriggersCompiledRules
+	Details               ClientTransportDetail `yaml:"Details"`
+}
+
+// Transports map with Transport
+type Transports []Transport
+
+// compileRule
+func (t *Transport) compileRule(regexpRule string) (compiledRule *regexp.Regexp, err error) {
+	if len(regexpRule) > 0 {
+		compiledRule, err = regexp.Compile(regexpRule)
+	}
+	return
+}
+
+// transportFlags for internal matching func
+type transportFlags struct {
+	declared bool
+	matched  bool
+	empty    bool
+}
+
+// compileRules prepares precompiled regular expressions for rules
+func (t *Transport) compileRules() error {
+	if !t.TriggersCompiledRules.IsCompiled {
+		if len(t.Triggers.Method) > 0 {
+			var err error
+			t.TriggersCompiledRules.MethodRegexp, err = t.compileRule(t.Triggers.Method)
+			if err != nil {
+				return fmt.Errorf("compileRule for Client->Transport->Trigger->Method error: %q", err)
+			}
+		}
+		if len(t.Triggers.Path) > 0 {
+			var err error
+			t.TriggersCompiledRules.PathRegexp, err = t.compileRule(t.Triggers.Path)
+			if err != nil {
+				return fmt.Errorf("compileRule for Client->Transport->Trigger->Path error: %q", err)
+			}
+		}
+		if len(t.Triggers.QueryParam) > 0 {
+			var err error
+			t.TriggersCompiledRules.QueryParamRegexp, err = t.compileRule(t.Triggers.QueryParam)
+			if err != nil {
+				return fmt.Errorf("compileRule for Client->Transport->Trigger->QueryParam error: %q", err)
+			}
+		}
+		t.TriggersCompiledRules.IsCompiled = true
+	}
+	return nil
+}
+
+// GetMatchedTransport returns first details matching with rules from Triggers by arguments: method, path, queryParam
+func (t *Transports) GetMatchedTransport(method, path, queryParam string) (defaultTransport Transport, defaultTransportName string, ok bool) {
+	for _, transport := range *t {
+		transport.compileRules()
+		methodFlag, pathFlag, queryParamFlag := matchTransportFlags(transport, method, path, queryParam)
+
+		if methodFlag.matched && pathFlag.matched && queryParamFlag.matched {
+			return transport, transport.Name, true
+		}
+		if methodFlag.empty && pathFlag.empty && queryParamFlag.empty && len(defaultTransportName) == 0 {
+			defaultTransport = transport
+			defaultTransportName = transport.Name
+			ok = true
+		}
+	}
+	return
```

Comment:
What for the `ok` return value, if it's always `true`?

Reply:
Done